### DISCUSSION TITLE
ViewModel Binding

### DISCUSSION THREAD

**Discussion by linusaurus** [lepoco/wpfui#1074] [General]
I dont understand why the DataContext is set to this rather than the specific viewmodel. Am I missing something. I spent many a hour trying to get the data from teh veiwmodel becuase it never set to the view using the example code. 

public partial class PartPage : INavigableView<PartViewModel>
 {

     public PartViewModel ViewModel { get; }
     public PartPage(PartViewModel viewModel)
     {
         ViewModel = viewModel;
         DataContext = viewModel;  // this was set to this in the examples
  
         InitializeComponent();
     }

    
 }`

**Comment by pomianowski** [comment]
This is simply how the Gallery app is made, it has nothing to do with WPF-UI. You can set the `DataContext` however you want.
